common/djangoapps/student/roles.py

```diff
@@ -47,9 +47,11 @@ def has_role(self, role, course_id, org):
         """
         Return whether this RoleCache contains a role with the specified role, course_id, and org
         """
+        # Convert course keys to unicode string before comparison. this is because some objects
+        # like ccx locator dont have overridden __eq__. Thats why when we compare them always return false.
         return any(
             access_role.role == role and
-            access_role.course_id == course_id and
+            unicode(access_role.course_id) == unicode(course_id) and
```

Comment:
Because code was not able to compare ccxLocator objects though it was able to compare courseLocator.
In sum when this line was comparing ccx keys it always give false even both key are match. Thats why I am deserialising keys and comparing them as strings.

second option is block this PR and create small PR against 
https://github.com/edx/ccx-keys/blob/master/ccx_keys/locator.py

and override eq method. Because we dont have this method thats why when i compare 2 ccx locator objects it always returns false

After that resume this PR
http://stackoverflow.com/questions/1227121/compare-object-instances-for-equality-by-their-attributes-in-python